sdk/dotnet/Pulumi/Deployment/Deployment_Inline.cs

```diff
@@ -38,7 +38,11 @@ private Deployment(InlineDeploymentSettings settings)
             Monitor = new GrpcMonitor(settings.MonitorAddr);
             deploymentLogger.LogDebug("Created deployment monitor");
 
-            _runner = new Runner(this, deploymentLogger);
+            // Tell the runner that we are running inside an inline automation program
```

Comment:
Does this mean that inline deployments that don't "return await" correctly can just hit the same issue this is trying to fix for normal deployments?